test/regression/api/canary/admin/redirects_spec.js

```diff
@@ -225,4 +252,132 @@ describe('Redirects API', function () {
             });
         });
     });
+
+    describe('Upload yaml', function () {
+        // No error cases here because there are no easy syntax pitfalls in the yaml format.
+
+        describe('Ensure re-registering redirects works', function () {
+            const startGhost = (options) => {
+                return ghost(options)
+                    .then(() => {
+                        request = supertest.agent(config.get('url'));
+                    })
+                    .then(() => {
+                        return localUtils.doAuth(request);
+                    });
+            };
+
+            it('no redirects file exists', function () {
+                return startGhost({redirectsFile: false, forceStart: true})
+                    .then(() => {
+                        return request
+                            .get('/my-old-blog-post/')
+                            .expect(404);
+                    })
+                    .then(() => {
+                        // Provide a redirects file in the root directory of the content test folder
+                        fs.writeFileSync(path.join(config.get('paths:contentPath'), 'redirects-init.yaml'), '302:\n  k: l');
+                    })
+                    .then(() => {
+                        return request
+                            .post(localUtils.API.getApiQuery('redirects/json/'))
```

Comment:
We'd probably want to change the endpoint name to something like `redirects/upload` and `redirects/download` instead of keeping the format name here. That would be similar to the naming use for members upload/download endpoints -  https://github.com/TryGhost/Ghost/blob/635580f291bd826c9119049ad968cb7702477ba7/core/server/web/api/canary/admin/routes.js#L95-L96

Reply:
Closing this in favor of the comment below.